### PR TITLE
automation_initiator_location set by user

### PR DESCRIPTION
**Pull request by samidbb**
## Describe your changes
Enable proper monitoring and adhere to tag policy

## Issue ticket number and link
-

## Checklist before requesting a review
- [x] I have tested changes in my sandbox
- [x] I have added the needed changes in the `test/qa` folder to apply my changes in QA.
- [x] I have rebased the code to main (or merged in the latest from main)

## Is it a new release?
- [x] Apply a release tag `release:(major|minor|patch)`, following semantic versioning in [this guide](https://semver.org/) or `norelease` if there is no changes to the Terraform code

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -817,3 +817,17 @@ EOF
...
+  description = <<EOF
+    Specify the URL to the repo of automation script.
+    Valid Values: URL to repo. Example: `"https://github.com/dfds/terraform-aws-rds"`
+    Notes: This set the dfds.automation.initiator.location tag. See recommendations [here](https://wiki.dfds.cloud/en/playbooks/standards/tagging_policy).
+  EOF
+  type        = string
+  default     = null
+  validation {
+    condition     = var.automation_initiator_location == null || can(regex("(https:\\/\\/www\\.|http:\\/\\/www\\.|https:\\/\\/|http:\\/\\/)?[a-zA-Z0-9]{2,}(\\.[a-zA-Z0-9]{2,})(\\.[a-zA-Z0-9]{2,})?\\/[a-zA-Z0-9]{2,}", var.automation_initiator_location))
+    error_message = "Invalid value for var.automation_initiator_location. Supported values: URL path. Example: https://github.com/dfds/terraform-aws-rds"
```
Comment (avnes) [inline variables.tf:831]: I think we need to extend the error message for ADO too since it won't work with `?` in the URL.

**Review by avnes — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (samidbb) [reply variables.tf:831]: hmm do have an example of such url that I can use to test?

Comment (avnes) [reply variables.tf:831]: Example sent on Slack

File: tools/scaffolding/templates/main.tf.template [hunk 2]
```diff
@@ -15,7 +15,7 @@ provider "aws" {
 
 
 module "db_instance" {
-  source = "git::https://github.com/dfds/terraform-aws-rds.git?ref=<release_number>"
+  source = "git::https://github.com/dfds/terraform-aws-rds.git?ref=0.0.1"
```
Comment (avnes) [inline tools/scaffolding/templates/main.tf.template:18]: We are already at 0.16.0

File: variables.tf [hunk 3]
```diff
@@ -804,16 +804,30 @@ EOF
...
     Valid Values: URL to repo. Example: `"https://github.com/dfds/terraform-aws-rds/actions/workflows/qa.yml"`
     Notes: This set the dfds.automation.initiator.pipeline tag. See recommendations [here](https://wiki.dfds.cloud/en/playbooks/standards/tagging_policy).
 EOF
   type        = string
   default     = null
   validation {
-    condition     = var.pipeline_location == null || can(regex("^(https:\\/\\/www\\.|http:\\/\\/www\\.|https:\\/\\/|http:\\/\\/)?[a-zA-Z0-9]{2,}(\\.[a-zA-Z0-9]{2,})([a-z0-9_@\\-^!#$%&+={}.\\/\\\\[\\]]+)+\\.(yml|yaml|sh)$", var.pipeline_location))
-    error_message = "Invalid value for var.pipeline_location. Supported values: URL path. Example: https://github.com/dfds/terraform-aws-rds/actions/workflows/qa.yml"
+    condition     = var.pipeline_location == null || can(regex("^(https:\\/\\/www\\.|http:\\/\\/www\\.|https:\\/\\/|http:\\/\\/)?[a-zA-Z0-9]{2,}(\\.[a-zA-Z0-9]{2,})(\\.[a-zA-Z0-9]{2,})?(\\/[a-zA-Z0-9_.:/=+-@][^?|^/&]{2,})+$", var.pipeline_location))
+    error_message = "Value for var.pipeline_location contains invalid characters. See AWS [user guide](https://docs.aws.amazon.com/tag-editor/latest/userguide/tagging.html) for more information."
```
Comment (avnes) [inline variables.tf:817]: Very nice!

**Review by avnes — CHANGES_REQUESTED** [review]

File: tools/scaffolding/templates/main.tf.template (hunk 2 above)
Comment (samidbb) [reply tools/scaffolding/templates/main.tf.template:18]: I know but I want to check if renovate can act on it.

**Review by avnes — APPROVED** [review]

**Pull request merged**